sdk/dotnet/Pulumi/Core/Input.cs

```diff
@@ -48,4 +49,31 @@ public Output<T> ToOutput()
         IOutput IInput.ToOutput()
             => ToOutput();
     }
+
+    public static class InputExtensions
+    {
+        /// <summary>
+        /// <see cref="Output{T}.Apply{U}(Func{T, Output{U}?})"/> for more details.
+        /// </summary>
+        public static Output<U> Apply<T, U>(this Input<T>? input, Func<T, U> func)
+            => input?.ToOutput().Apply(func) ?? Output.Create(default(U)!);
+
+        /// <summary>
+        /// <see cref="Output{T}.Apply{U}(Func{T, Output{U}?})"/> for more details.
+        /// </summary>
+        public static Output<U> Apply<T, U>(this Input<T>? input, Func<T, Task<U>> func)
+            => input?.ToOutput().Apply(func) ?? Output.Create(default(U)!);
+
+        /// <summary>
+        /// <see cref="Output{T}.Apply{U}(Func{T, Output{U}?})"/> for more details.
+        /// </summary>
+        public static Output<U> Apply<T, U>(this Input<T>? input, Func<T, Input<U>?> func)
+            => input?.ToOutput().Apply(func) ?? Output.Create(default(U)!);
+
+        /// <summary>
+        /// <see cref="Output{T}.Apply{U}(Func{T, Output{U}?})"/> for more details.
+        /// </summary>
+        public static Output<U> Apply<T, U>(this Input<T>? input, Func<T, Output<U>?> func)
+            => input?.ToOutput().Apply(func) ?? Output.Create(default(U)!);
```

Comment:
can now call `.Apply` on any input (even a null one) to get back a reasonable non-null Output.